base/hl7_client.bal

```diff
@@ -37,8 +38,17 @@ public class HL7Client {
             anydata mshSegment = message.get("msh");
             if mshSegment is Segment {
                 anydata hl7Version = mshSegment.get("msh12");
+                // from hl7v24 onwards, MSH12 has 3 parts vid1(version id), vid2(internationalization id), vid3(internal version id)
+                string|error versionId = "";
                 if hl7Version is string {
-                    byte[]|HL7Error encodedMessage = encode(hl7Version, message);
+                    versionId = hl7Version;
+                } else {
+                    json vid = hl7Version.toJson();
```

Comment:
Let's have vid record created in base and use it? Also handle the error scenario.

Reply:
fixed in [1df97fd](https://github.com/ballerina-platform/module-ballerinax-health.hl7v2/pull/242/commits/1df97fde7d08146ae0d3f944e0afc9a1d0190da5)